maas-api/internal/tier/mapper.go

```diff
@@ -1,38 +1,53 @@
 package tier
 
 import (
-	"context"
 	"errors"
 	"fmt"
 	"log"
 	"slices"
 	"sort"
 	"strings"
+	"time"
 
 	"gopkg.in/yaml.v3"
 	k8serrors "k8s.io/apimachinery/pkg/api/errors"
-	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/informers"
 	"k8s.io/client-go/kubernetes"
-	corev1typed "k8s.io/client-go/kubernetes/typed/core/v1"
+	corelisters "k8s.io/client-go/listers/core/v1"
 
 	"github.com/opendatahub-io/maas-billing/maas-api/internal/constant"
 )
 
 // Mapper handles tier-to-group mapping lookups.
 type Mapper struct {
 	tenantName      string
-	configMapClient corev1typed.ConfigMapInterface
+	namespace       string
+	configMapLister corelisters.ConfigMapLister
 }
 
 func NewMapper(clientset kubernetes.Interface, tenantName, namespace string) *Mapper {
+	informerFactory := informers.NewSharedInformerFactoryWithOptions(
+		clientset,
+		30*time.Second,
```

Comment:
This could even be O(hours)

Reply:
That makes sense, I think we can align with ctrl-runtime default resync to be 8h. I will fix it in other places too.